### PR TITLE
feat(conditionalIntl): added env var for Itnl

### PR DESCRIPTION
**Pull request by timBarton96**
## Description
Added an environment variable to conditionally add the lean-intl polyfill.

## Motivation and Context
The lean-intl polyfill is missing Intl fetres and this PR is for consumers of one-app that want to trade compatibility with older browsers for these Intl features. The environment variable defaults to enable lean-intl polyfill, so needs to be explicitly be disabled. By disalbing the polyfill, the Intl implementations of the browsers are used and for modern browsers, this means extended functionality. 

## How Has This Been Tested?
I served a test-module which uses Luxon which uses Intl for its timezones functionality. The timezones would not work without the env var, but when the env var is enabled and there is no lean-intl the timezones work.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [x] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using One App?
Developers will now have the option to opt out of using lean-intl.

### CODE REVIEW

File: docs/api/server/Environment-Variables.md [hunk 1]
```diff
@@ -524,6 +527,27 @@ ONE_CONFIG_ENV=staging
 ONE_CONFIG_ENV=undefined
 ```
 
+## `ONE_CONFIG_USE_NATIVE_POLYFILL`
```
Comment (Matthew-Mallimo) [inline docs/api/server/Environment-Variables.md:530]:
```suggestion
## `ONE_CONFIG_USE_NATIVE_INTL`
```

I think this makes it a bit clearer what the variable does

File: src/server/plugins/reactHtml/index.jsx [hunk 2]
```diff
@@ -236,6 +236,12 @@ export function getHead({
   `;
 }
 
+export function renderUseNativeIntlPolyfill(nonce) {
+  return process.env.ONE_CONFIG_USE_NATIVE_POLYFILL === 'true' ? `<script id="environment-variables" ${nonce}>
+        window.useNativePolyfill = 'true'
+       </script>` : '';
+}
```
Comment (Matthew-Mallimo) [inline src/server/plugins/reactHtml/index.jsx:243]:
```suggestion
export function renderEnvironmentVariables(nonce) {
  return `<script id="environment-variables" ${nonce}>
        window.useNativePolyfill = ${process.env.ONE_CONFIG_USE_NATIVE_POLYFILL === 'true'}
       </script>`;
}
```

I think making this function a bit more general is better. It sets us up to add more env vars in the future

**Review by Matthew-Mallimo — CHANGES_REQUESTED** [review]

File: src/server/plugins/reactHtml/index.jsx [hunk 3]
```diff
@@ -268,6 +275,7 @@ export function getBody({
         window.__render_mode__ = '${renderMode}';
         window.__HOLOCRON_EXTERNALS__ = ${jsonStringifyForScript(getRequiredExternalsRegistry())};
       </script>
+      ${renderUseNativeIntlPolyfill(nonce)}
       ${assets}
       ${renderI18nScript(clientInitialState, bundlePrefixForBrowser)}
```
Comment (10xLaCroixDrinker) [inline src/server/plugins/reactHtml/index.jsx:280]: If using native Intl, this file should not be included

File: src/server/polyfill/intl.js [hunk 4]
```diff
@@ -14,4 +14,8 @@
  * permissions and limitations under the License.
  */
 
-global.Intl = require('lean-intl');
+import Intl from 'lean-intl';
+
+if (!process.env.ONE_CONFIG_USE_NATIVE_POLYFILL) {
```
Comment (10xLaCroixDrinker) [inline src/server/polyfill/intl.js:19]:
```suggestion
if (process.env.ONE_CONFIG_USE_NATIVE_POLYFILL !== 'true') {
```

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]
This also requires changes to `one-app-ducks` to prevent loading intl polyfill locale packs when using native Intl https://github.com/americanexpress/one-app-ducks/blob/main/src/intl/index.js#L436

File: docs/api/server/Environment-Variables.md (hunk 1 above)
Comment (timBarton96) [reply docs/api/server/Environment-Variables.md]: That's a good point, I have updated the name to this in [d76bbe5](https://github.com/americanexpress/one-app/pull/1258/commits/d76bbe516d1724bdd802457057f5727466429a71)

File: src/server/plugins/reactHtml/index.jsx (hunk 2 above)
Comment (timBarton96) [reply src/server/plugins/reactHtml/index.jsx]: I have made this function more generic in [d76bbe5](https://github.com/americanexpress/one-app/pull/1258/commits/d76bbe516d1724bdd802457057f5727466429a71)

File: src/server/polyfill/intl.js (hunk 4 above)
Comment (timBarton96) [reply src/server/polyfill/intl.js]: Good catch! I have fixed this in [d76bbe5](https://github.com/americanexpress/one-app/pull/1258/commits/d76bbe516d1724bdd802457057f5727466429a71)

File: src/server/plugins/reactHtml/index.jsx (hunk 3 above)
Comment (timBarton96) [reply src/server/plugins/reactHtml/index.jsx:284]: As discussed, I have added a guard clause to this function to return early if the env var equals 'true' in [edb789a](https://github.com/americanexpress/one-app/pull/1258/commits/edb789aa4e5e79e34d18e794e45fc2c3de99841f)

**Review by Matthew-Mallimo — DISMISSED** [review]

**Comment by Matthew-Mallimo** [comment]
@timBarton96  [one-app-ducks@4.5.0](https://github.com/americanexpress/one-app-ducks/releases/tag/v4.5.0) has been released. Please include this upgrade in this PR!

**Review by Matthew-Mallimo — DISMISSED** [review]

File: docs/api/server/Environment-Variables.md [hunk 5]
```diff
@@ -29,13 +29,15 @@ One App can be configured via Environment Variables:
...
   * [`ONE_CONFIG_ENV`](#one_config_env) ⚠️
+  * [`ONE_CONFIG_USE_NATIVE_INTL`](#one_config_use_native_intl)
 * Running in Development
   * [`NODE_ENV`](#node_env) ⚠️
   * [`ONE_CLIENT_ROOT_MODULE_NAME`](#one_client_root_module_name) ⚠️
   * [`ONE_CONFIG_ENV`](#one_config_env) ⚠️
   * [`ONE_DANGEROUSLY_ACCEPT_BREAKING_EXTERNALS`](#ONE_DANGEROUSLY_ACCEPT_BREAKING_EXTERNALS)
   * [`ONE_CSP_ALLOW_INLINE_SCRIPTS`](#ONE_CSP_ALLOW_INLINE_SCRIPTS)
   * [`ONE_DANGEROUSLY_DISABLE_CSP`](#ONE_DANGEROUSLY_DISABLE_CSP)
+  * * [`ONE_CONFIG_USE_NATIVE_INTL`](#one_config_use_native_intl)
```
Comment (10xLaCroixDrinker) [inline docs/api/server/Environment-Variables.md:40]:
```suggestion
  * [`ONE_CONFIG_USE_NATIVE_INTL`](#one_config_use_native_intl)
```

**Review by 10xLaCroixDrinker — CHANGES_REQUESTED** [review]

File: docs/api/server/Environment-Variables.md (hunk 5 above)
Comment (timBarton96) [reply docs/api/server/Environment-Variables.md]: Sure, fixed in [65a6992](https://github.com/americanexpress/one-app/pull/1258/commits/65a699202bdf803beb95baac0a58f2a5c4e24ae7)

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by 10xLaCroixDrinker — DISMISSED** [review]

File: __tests__/integration/helpers/testRunner.js [hunk 6]
```diff
@@ -95,7 +95,7 @@ const setUpTestRunner = async ({
   }));
 
   // uncomment this line in order to view full logs for debugging
-  // logWatcherDuplex.pipe(process.stdout);
+  logWatcherDuplex.pipe(process.stdout);
```
Comment (Matthew-Mallimo) [inline __tests__/integration/helpers/testRunner.js:98]:
```suggestion
 //  logWatcherDuplex.pipe(process.stdout);
```

Comment (code-forger) [reply __tests__/integration/helpers/testRunner.js:98]: I'm personally keen on keeping this like this.

When would you be looking at the integration test logs, and not want the pod logs right there?

**Review by Matthew-Mallimo — APPROVED** [review]

**Review by code-forger — APPROVED** [review]

File: src/server/plugins/reactHtml/index.jsx [hunk 7]
```diff
@@ -236,6 +240,12 @@ export function getHead({
   `;
 }
 
+export function renderEnvironmentVariables(nonce) {
```
Comment (JAdshead) [inline src/server/plugins/reactHtml/index.jsx:243]: just a naming nitpick, `renderEnvironmentVariables` indicates that its a standard practice to include, right now this is okay but could lead to issues in the future

**Pull request merged**